source/DEVS-Network/DEVSUDPServer.class.st

```diff
@@ -0,0 +1,68 @@
+Class {
+	#name : 'DEVSUDPServer',
+	#superclass : 'DEVSAtomicComponent',
+	#instVars : [
+		'state',
+		'address',
+		'messages'
+	],
+	#category : 'DEVS-Network',
+	#package : 'DEVS-Network'
+}
+
+{ #category : 'accessing' }
+DEVSUDPServer >> address [
+	^ address
+]
+
+{ #category : 'accessing' }
+DEVSUDPServer >> address: aString [ 
+	address := aString.
+	(address endsWith: '1') ifTrue: [ 
+		messages add: (DEVSNetworkMessage new 
+				from: name;
+				to: 'server3';
+				payload: #foo)  ]
+]
+
+{ #category : 'modeling' }
+DEVSUDPServer >> externalTransition [ 
+	| msg |
+	msg := self peekFrom: #in.
+	(msg from = '10.0.0.1') ifTrue: [  
+	messages add: (DEVSNetworkMessage new
+		from: msg to;
+		to: msg from;
+		payload: #bar) ].
+
+]
+
+{ #category : 'modeling' }
+DEVSUDPServer >> initialize [ 
+	super initialize.
+	self addInputPortNamed: #in.
+	self addOutputPortNamed: #out.
+	state := #idle.
```

Comment:
state variable is not used anywhere in the code (It seems to be useless here).

Reply:
Yes this code is in "make it work" mode so not final. I usually hack away until it makes sense and then it gets refactored to be proper